pkg/alertmanager/operator.go

```diff
@@ -1490,6 +1495,41 @@ func checkSnsConfigs(
 	return nil
 }
 
+func checkTelegramConfigs(
+	ctx context.Context,
+	configs []monitoringv1alpha1.TelegramConfig,
+	namespace string,
+	key string,
+	store *assets.Store,
+	amVersion semver.Version,
+) error {
+	for i, config := range configs {
```

Comment:
I think we will need to check the alertmanager version here for compatibility reasons since this integration was only added in `0.24`

See [the route check](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/alertmanager/operator.go#L1134-L1139) for example!

Reply:
You're totally right, I'll take a look!